build/features.py

```diff
@@ -812,13 +812,30 @@ def enabled(self, build):
 
     def add_options(self, build, vars):
         vars.Add('shoutcast', 'Set to 1 to enable live broadcasting support', 1)
+	vars.Add('shoutcast_internal', 'Set to 1 to use internal libshout', 1)
 
     def configure(self, build, conf):
         if not self.enabled(build):
             return
 
-        libshout_found = conf.CheckLib(['libshout', 'shout'])
         build.env.Append(CPPDEFINES='__BROADCAST__')
+        
+        build.flags['shoutcast_internal'] = util.get_flags(build.env, 'shoutcast_internal', 0)        
+        if build.platform_is_linux and not build.flags['shoutcast_internal']:
+            # Check if system lib is lower 2.4.2 or 2.4.3 and not suffering bug 
+            # https://bugs.launchpad.net/mixxx/+bug/1833225
+            if conf.CheckForPKG('shout', '2.3.1'):
```

Comment:
Why do you check for version 2.3.1 if the issue is only present in 2.4.2 and 2.4.3?

I read through the linked issue and to me it looks like the issue has been fixed in `master` in this commit: https://gitlab.xiph.org/xiph/icecast-libshout/commit/0ac7ed9e84c3871d4427acc1ce59dca5e4af21ef

I'm sure this fix will be present in the next release of libshout. Can we do a check like this: if version >= 2.4.2 and version <= 2.4.3 then use internal libshout by default, else use system libshout?

Reply:
Ups, that was a test with my system libshout which is 2.3.1